lib/src/lib.rs

```diff
@@ -222,9 +220,53 @@ impl Tester {
                     }
                 };
 
+                log::info!("Testing Actor {}", target.name);
+
                 let root = self.state_tree.flush();
+                let blockstore = self.state_tree.store().clone();
+                let mut executor = Self::new_executor(blockstore, root, self.builtin_actors.root);
+
+                // Run the constructor if it exists.
+                if let Some(constructor) = test_actor.abi().constructor() {
+                    let message = Message {
+                        from: self.account.1,
+                        to: test_address,
+                        gas_limit: 1000000000,
+                        method_num: constructor.number(),
+                        params: target_id.clone().into(),
+                        ..Message::default()
+                    };
+                    if let Err(err) = executor
+                        .execute_message(message, ApplyKind::Explicit, 100)
+                        .setting_err("Could not run Constructor")
+                    {
+                        return TestActorResults {
+                            test_actor,
+                            results: Err(err),
+                        };
+                    }
+                }
 
-                log::info!("Testing Actor {}", target.name);
+                // Run SetUp if it exists.
+                if let Some(set_up) = test_actor.abi().set_up() {
+                    let message = Message {
+                        from: self.account.1,
+                        to: test_address,
+                        gas_limit: 1000000000,
+                        method_num: set_up.number(),
+                        params: target_id.clone().into(),
+                        ..Message::default()
+                    };
```

Comment:
The problem from the tests comes from here. As it goes we send messages with the same sequence (nonce) everytime. The sequence should be incremented each time we send a new explicit message.

This can be seen in return of the execution:
```
Failed(
    ApplyRet {
        msg_receipt: Receipt {
            exit_code: ExitCode {
                value: 2,
            },
            return_data: RawBytes {  },
            gas_used: 0,
            events_root: None,
        },
        penalty: TokenAmount(0.0000001),
        miner_tip: TokenAmount(0.0),
        base_fee_burn: TokenAmount(0.0),
        over_estimation_burn: TokenAmount(0.0),
        refund: TokenAmount(0.0),
        gas_refund: 0,
        gas_burned: 0,
        failure_info: Some(
            PreValidation(
                "Actor sequence invalid: 0 != 1",
            ),
        ),
        exec_trace: [],
        events: [],
    },
)

```